doc/neps/nep-0045-c_style_guide.rst

```diff
@@ -0,0 +1,261 @@
+=================================
```

Comment:
Maybe put `.. _NEP45` up top to ease linking to this document. There is also a reference in `doc/source/dev/style_guide.rst` that needs a fix, not sure of the best way to do that.

Reply:
https://docutils.sourceforge.io/docs/ref/rst/directives.html#including-an-external-document-fragment explains the `include` directive, that may help.

Reply:
`doc/source/dev/style_guide.rst` can be removed, and `source/dev/index` can be changed to link to the NEP. 

I suggest to directly link to the absolute URL (`https://numpy.org/neps/nep-0045-...`), because the `_NEP45` anchor thing won't work well probably, given that the `doc/neps` is a completely separate Sphinx build and deploy.

Reply:
Do you know if the links are checked in the CircleCI builds?

Reply:
Not yet I believe, because we're not at zero warnings yet so can't turn on the check.

Easy to verify manually though via the ` ci/circleci: build artifact` link in the CI status list.

Reply:
I ran a checker when preparing my Season of Docs proposal. Nearly every page has a bad link because of the 'Created using Sphinx 2.4.4' line at the bottom. It links to `http://sphinx.pocoo.org/`, which returns a 400.

I found about two dozen other fails. Starting a PR.